### PR TITLE
Supporting generic *-dummy services for testing

### PR DESCRIPTION
**Pull request by giorgiosironi**
Extracted from `profiles.orcid-dummy`.

### CODE REVIEW

File: elife/php-dummies.sls [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        - name: git checkout $(cat {{ configuration['pinned_revision_file'] }})
+        - cwd: /srv/{{ name }}
+        - user: {{ pillar.elife.deploy_user.username }}
+        - require:
+            - file: {{ name }}-repository
+
+{{ name }}-composer-install:
+    cmd.run:
+        {% if pillar.elife.env != 'dev' %}
+        - name: composer --no-interaction install --no-suggest --classmap-authoritative
```
Comment (thewilkybarkid) [inline elife/php-dummies.sls]: `--no-dev`?

File: elife/php-dummies.sls [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        {% else %}
+        - name: composer --no-interaction install --no-suggest
```
Comment (thewilkybarkid) [inline elife/php-dummies.sls]: Probably ok to separate `dev`, you might want to tweak the dummy at the same time as the main project.

Comment (giorgiosironi) [reply elife/php-dummies.sls]: Least astonishment, ok

**Comment by giorgiosironi** [comment]
Well, composer and the nginx setup are PHP-specific so trying to use it with other dummies will break.

**Pull request merged**